### PR TITLE
Implementing grunt

### LINKED ISSUE #23: Add Wordpress publishing
replacement issue of #13

### PR DESCRIPTION
**Pull request by dutchmartin**
This PR contains:

- Grunt commands to create an Artifact folder (+zip)
- Grunt commands to version bump the plugin
- A new readme.txt, which is currently empty on purpose 
- The `grunt deploy:master` command
- The `grunt deploy:trunk` command
Test instructions:
Ensure you have all packages installed by running `yarn`.
Run the ` grunt artifact:zip` command, this should generate a new artifact zip.

Fixes #23

### CODE REVIEW

**Comment by moorscode** [comment]
Missing the `vendor` folder.
This should be generated in the same manner as done in the WordPress-seo repository.
Running a composer install --no-dev and copying the vendor folder when creating the artifact.

**Comment by dutchmartin** [comment]
Grunt artifact runs a composer no dev command, but I have added the feature that reverts that after the artifact generation so you can continue developing.

File: Gruntfile.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const path = require( "path" );
+const loadGruntConfig = require( "load-grunt-config" );
+
+module.exports = function (grunt) {
```
Comment (jcomack) [inline Gruntfile.js]: Please properly apply code-style.

Take the following things into account:

* Spaces after and before `(` and `)`.
* Usage of `'` and `"`.

File: grunt/config/set-version.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// Custom task
```
Comment (jcomack) [inline grunt/config/set-version.js]: Could you please add proper descriptions for the task.

File: grunt/config/update-version.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+// Custom task
```
Comment (jcomack) [inline grunt/config/update-version.js]: Could you please add proper descriptions for the task.

File: grunt/config/update-version.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        options: {
+            regEx: /(Stable tag:\s+)(\d+(\.\d+){0,3})([^\n^\.\d]?.*?)(\n)/,
+            preVersionMatch: "$1",
+            postVersionMatch: "$5",
+        },
+        src: "readme.txt",
+    },
+    packageJSON: {
+        options: {
+            // "version": "1.0.0",
```
Comment (jcomack) [inline grunt/config/update-version.js]: Is there a reason for this commented out property? If not, please remove it.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "yoast-test-helper",
+  "version": "1.0.2",
+  "description": "A plugin to aid in testing and developing the Yoast SEO plugin and its extensions.",
+  "repository": "https://github.com/Yoast/yoast-test-helper.git",
+  "author": "Yoast",
+  "license": "SEE LICENSE IN licence.txt",
```
Comment (jcomack) [inline package.json]: Shouldn't this be `license.txt`?

Comment (dutchmartin) [reply package.json]: npm documentation says "{ "license" : "SEE LICENSE IN <filename>" }": https://docs.npmjs.com/files/package.json

Comment (jcomack) [reply package.json]: I get that. What I'm saying is that the filename `licence.txt` is spelled incorrectly if you consider we use American spelling.

**Comment by jcomack** [comment]
CR done 👍

**Comment by moorscode** [comment]
Can you resolve the merge conflicts please?

**Comment by dutchmartin** [comment]
This pr can be squashed and merged

**Pull request merged**